### ISSUE TITLE
getting started

### ISSUE THREAD

**Issue by nilsbecker** [thierry-martinez/pyml#1]
it would be great to have some introductory examples just to get started with something trivial. i tried 
```
Py.initialize ()
Py.Run.ipython ()
```
but this gave an exception 
```
exception E (<type 'exceptions.ImportError'>, dlopen(/Users/nbecker/Apps/anaconda/lib/python2.7/lib-dynload/_io.so, 2): Symbol not found: __PyCodecInfo_GetIncrementalDecoder
  Referenced from: /Users/nbecker/Apps/anaconda/lib/python2.7/lib-dynload/_io.so
  Expected in: flat namespace
 in /Users/nbecker/Apps/anaconda/lib/python2.7/lib-dynload/_io.so)
```
probably because i did not do it in the right way. i think some trivial examples could help.

**Comment by thierry-martinez** [comment]
Such commands look fine and I get a working IPython session after the
commands you execute: I cannot reproduce the exception that you
observed, nor the segfaults you mention on another email (tested on
macOS Sierra, Yosemite and Ubuntu 16.04).

Did you try with master?
`opam pin add pyml http://pyml.gforge.inria.fr/pyml.git`

**Comment by nilsbecker** [comment]
yes this was from master... i get the same in the toplevel i built within the pyml build directory using make, (but there without printing of the exception) and from a simple compiled program.
os x 10.11, python 2.7.11 from anaconda (which is first in the path, not the system python)

**Comment by nilsbecker** [comment]
i'm using the +flambda compiler switch but i guess that should not matter for ocaml bytecode?

**Comment by thierry-martinez** [comment]
Can you try to execute `export PYTHONHOME=/Users/nbecker/Apps/anaconda/` before executing `pymltop`? Not sure how to fix that properly yet (I don’t know how to guess a proper value for `PYTHONHOME` in every context…).

**Comment by nilsbecker** [comment]
i tried; still get the same exception unfortunately:
```
Exception:
Py.E (<type 'exceptions.ImportError'>,
 dlopen(/Users/nbecker/Apps/anaconda/lib/python2.7/lib-dynload/_io.so, 2): Symbol not found: __PyCodecInfo_GetIncrementalDecoder
  Referenced from: /Users/nbecker/Apps/anaconda/lib/python2.7/lib-dynload/_io.so
  Expected in: flat namespace
 in /Users/nbecker/Apps/anaconda/lib/python2.7/lib-dynload/_io.so).
#
```

**Comment by thierry-martinez** [comment]
And `export DYLD_LIBRARY_PATH="/Users/nbecker/Apps/anaconda/lib"`? Thanks.

**Comment by nilsbecker** [comment]
actually i have `PYTHONPATH=/usr/local/lib/python:`, maybe this is a problem. will try without that as well

**Comment by nilsbecker** [comment]
ok i unset the pythonpath and set the dyld_library_path; unfortunately still the same error. i also have CAML_LD_LIBRARY_PATH set to something within opam/..../stublibs but i gues that's not important?

**Comment by nilsbecker** [comment]
my ipython is 4.0.3

**Comment by nilsbecker** [comment]
wait. now i got it working for the first time. i exported the PYTHONHOME and the DYLD_LIBRARY_PATH. will try again in a fresh shell to confirm

**Comment by nilsbecker** [comment]
yes. with both variables set, i can get a working ipython interactive session!

**Comment by nilsbecker** [comment]
so setting these is a working workaround. i was able to define python objects in the interactive ipython session and then read some of their attributes back in the ocaml toplevel -- i'm getting there. do i understand correctly that nevertheless forking the ipython session and having both run in parallel is not supported? (i can always switch back and forth with Run.ipython but having them in parallel would be a bit more comfortable)

**Comment by thierry-martinez** [comment]
Having to set PYTHONHOME and DYLD_LIBRARY_PATH is ugly, but I don’t see a cleaner way right now. Will keep thinking about that.

About having IPython and OCaml top-level in parallel: I don’t see how to do that, and I suppose that if it were possible, it would cause re-entrance problems. Cannot you register the OCaml functions you need to call as Python callable values before running IPython, and than call your OCaml functions from Python ?

**Comment by nilsbecker** [comment]
i guess i can, and this seems a workable solution - i was just asking to be sure. thanks.
